### PR TITLE
Updates api routes for camera plugin integrations

### PR DESCRIPTION
**Pull request by praffq**
## Proposed Changes

Updated routes for the camera plugin integration in backend.

backend pr : https://github.com/ohcnetwork/care/pull/2610

@ohcnetwork/care-fe-code-reviewers

## Summary by CodeRabbit

- **New Features**
	- Updated camera presets fetching mechanism to use external IDs for improved consistency.
	- Enhanced logic for managing camera presets in the `ConsultationFeedTab` component.

- **Bug Fixes**
	- Refined handling of preset states to ensure accurate retrieval based on external IDs.

- **Chores**
	- Streamlined API routing for camera presets, consolidating and renaming endpoints for clarity.

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]

File: src/components/CameraFeed/ConfigureCamera.tsx [hunk 1]
```diff
@@ -552,9 +557,11 @@ export default function ConfigureCamera(props: Props) {
                                   FeedRoutes.updatePreset,
                                   {
                                     pathParams: {
-                                      assetbed_id: selectedAssetBed.id,
                                       id: preset.id,
                                     },
+                                    query: {
+                                      assetbed_external_id: selectedAssetBed.id,
+                                    },
```
[the remark this answers is not in the document]
Comment (rithviknishad) [reply src/components/CameraFeed/ConfigureCamera.tsx]: PATCH request is being sent. so if it's undefined, it'd be kept unchanged right?

**Comment by nihal467** [comment]
@DraKen0009 can you check the coderabbit suggestion

CC: @rithviknishad

**Comment by rithviknishad** [comment]
@nihal467 not in scope of this PR; also `request` handles errors gracefully.

File: src/components/Facility/ConsultationDetails/ConsultationFeedTab.tsx [hunk 2]
```diff
@@ -113,11 +112,8 @@ export const ConsultationFeedTab = (props: ConsultationTabProps) => {
 
     const { data } = await operate({ type: "get_status" });
     const { position } = (data as { result: { position: PTZPayload } }).result;
-    const { data: updated } = await request(FeedRoutes.updatePreset, {
-      pathParams: {
-        assetbed_id: preset.asset_bed.id,
-        id: preset.id,
-      },
+    const { data: updated } = await request(FeedRoutes.positionPresets.update, {
+      pathParams: { id: preset.id },
```
[the remark this answers is not in the document]
Comment (rithviknishad) [reply src/components/Facility/ConsultationDetails/ConsultationFeedTab.tsx:106]: `operate` uses `request` utility fn. which does handle errors gracefully.

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
image
image

- [ ] the camera preset creation is not working and the related api is throwing 500

**Comment by rithviknishad** [comment]
Fixed in https://github.com/ohcnetwork/care_camera_asset/commit/ffe864b703b54a921db49a2f4c723f82a566fbfe

**Comment by nihal467** [comment]
image

@rithviknishad @DraKen0009 the camera preset is still not working, do give it a check before requesting next review

**Comment by nihal467** [comment]
@rithviknishad @DraKen0009 is it ready for review, I am seeing few commits in the backend

**Comment by praffq** [comment]
@nihal467  It need some more minor changes and some testing before final review. It should be done by tomorrow eod. I'll ping you once done

**Comment by nihal467** [comment]
@DraKen0009 what is the update on the PR

**Comment by praffq** [comment]
@nihal467  good for testing.

**Comment by nihal467** [comment]
image

seem like the camera is not working in local, the preset and stream api's are failing

image

the middleware is communication properly with the local, i have tested it with vital monitor

**Comment by praffq** [comment]
Add JWKS_BASE64 in /docker/local.env before building image . Should work that way

Also if possible , please share network tab logs

**Comment by nihal467** [comment]
> Add JWKS_BASE64 in /docker/local.env before building image . Should work that way
> 
> Also if possible , please share network tab logs

Its already added, seem like a middleware issue i guess , not working for local as well

image

**Comment by nihal467** [comment]
@khavinshankar make sure before the frontend merge, the cypress is passing

**Comment by rithviknishad** [comment]
we can re-open after rebasing with new changes and making FE too as a plugin

**Pull request closed without merging**